ui/src/app/cohort-search/search-bar/search-bar.component.tsx

```diff
@@ -0,0 +1,300 @@
+        if (!optionNames.includes(option.name)) {
+          optionNames.push(option.name);
+          return true;
+        }
+        return false;
+      });
+      this.setState({highlightedOption: null, loading: false, options});
+    }, (err) => this.setState({error: err}));
+  }
+
+  get showOverflow() {
+    const {options} = this.state;
+    return options && options.length <= 10;
+  }
+
+  selectOption(option: any) {
+    if (option) {
+      const {selectOption, setIngredients, setInput} = this.props;
+      setInput(option.name);
+      this.setState({highlightedOption: null, options: null, optionSelected: true});
+      if (option.type === CriteriaType.BRAND.toString()) {
+        const cdrId = +(currentWorkspaceStore.getValue().cdrVersionId);
+        cohortBuilderApi().getDrugIngredientByConceptId(cdrId, option.conceptId)
+          .then(resp => {
+            if (resp.items.length) {
+              const ingredients = resp.items.map(it => it.name);
+              setIngredients(ingredients);
+              // just grabbing the first one on the list for now
+              selectOption(resp.items[0]);
+            }
+          });
+      } else {
+        setIngredients(null);
+        selectOption(option);
+      }
+    }
+  }
+
+  onKeyDown(key: Key) {
+    switch (key) {
+      case Key.ArrowDown:
+        this.moveDown();
+        break;
+      case Key.ArrowUp:
+        this.moveUp();
+        break;
+      case Key.Enter:
+        this.enterSelect();
+    }
+  }
+
+  moveUp() {
+    const {highlightedOption} = this.state;
+    if (highlightedOption === 0) {
+      this.setState({highlightedOption: null});
+    } else if (highlightedOption > 0) {
+      this.setState({highlightedOption: highlightedOption - 1});
+    }
+  }
+
+  moveDown() {
+    const {highlightedOption, options} = this.state;
+    if (highlightedOption === null) {
+      this.setState({highlightedOption: 0});
+    } else if ((highlightedOption + 1) < options.length) {
+      this.setState({highlightedOption: highlightedOption + 1});
+    }
+  }
+
+  enterSelect() {
+    const {highlightedOption, options} = this.state;
+    this.selectOption(options[highlightedOption]);
+  }
+
+  render() {
+    const {highlightedOption, loading, options} = this.state;
+    const inputValue = highlightedOption !== null ? options[highlightedOption].name : this.props.searchTerms;
+    return <div style={{position: 'relative', width: '100%'}}>
+      <div style={styles.searchContainer}>
+        <div style={styles.searchBar}>
+          {loading ? <Spinner style={{verticalAlign: 'middle'}} size={16}/> : <ClrIcon shape='search' size='18'/>}
+          <TextInput style={styles.searchInput}
+            value={inputValue}
+            onChange={(e) => this.props.setInput(e)}
+            onKeyDown={(e) => this.onKeyDown(e.key)}/>
+        </div>
+      </div>
+      {options !== null && <div ref={(el) => this.dropdown = el} style={styles.dropdownMenu}>
```

Comment:
I think we have a dropdown component. Could you reuse that here? If not, what is stopping you? 

If we don't have a dropdown component, I'd love to make one.

Reply:
Yeah, I could easily see us needing to search other ontologies and highlight results (like for disease focused research on the workspace create page)